plugins/AddLeptonJetRelatedVariables.cc

```diff
@@ -0,0 +1,282 @@
+// constants, enums and typedefs
+//
+
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+AddLeptonJetRelatedVariables::AddLeptonJetRelatedVariables(const edm::ParameterSet& iConfig):
+  dRmax_(iConfig.getParameter<double>("dRmax")),
+  subLepFromJetForPtRel_(iConfig.getParameter<bool>("subLepFromJetForPtRel"))
+{
+  edm::InputTag jetcollection = iConfig.getParameter<edm::InputTag>("RawJetCollection");
+  jetCollectionTag_ = consumes<reco::PFJetCollection>(jetcollection);
+
+  edm::InputTag leptoncollection = iConfig.getParameter<edm::InputTag>("LeptonCollection");
+  leptonCollectionTag_ = consumes<reco::CandidateView>(leptoncollection);
+
+  edm::InputTag l1Cortag = iConfig.getParameter<edm::InputTag>("L1Corrector");
+  tagL1Corrector_ = consumes<reco::JetCorrector>(l1Cortag);
+
+  edm::InputTag l1l2l3ResCortag = iConfig.getParameter<edm::InputTag>("L1L2L3ResCorrector");
+  tagL1L2L3ResCorrector_ = consumes<reco::JetCorrector>(l1l2l3ResCortag);
+
+  edm::InputTag pfcandidatecoll = iConfig.getParameter<edm::InputTag>("pfCandidates");
+  pfCandidates_ = consumes<PFCollection>(pfcandidatecoll);
+  
+  bTagCollectionTag_ = consumes<reco::JetTagCollection>(edm::InputTag("pfCombinedInclusiveSecondaryVertexV2BJetTags"));
+  
+  vertexes_ = consumes<reco::VertexCollection>(edm::InputTag("offlinePrimaryVertices"));
+
+  //now do what ever initialization is needed
+  produces<edm::ValueMap<float> >("JetPtRatio");
+  produces<edm::ValueMap<float> >("JetPtRel");
+  produces<edm::ValueMap<float> >("JetNDauCharged");
+  produces<edm::ValueMap<float> >("JetBTagCSV");
+}
+
+AddLeptonJetRelatedVariables::~AddLeptonJetRelatedVariables()
+{
+
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+
+}
+
+
+//
+// member functions
+//
+
+
+
+
+// ------------ method called for each event  ------------
+void 
+AddLeptonJetRelatedVariables::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) 
+{
+  using namespace edm;
+
+  edm::Handle<reco::PFJetCollection> jets;       
+  iEvent.getByToken (jetCollectionTag_, jets);    
+
+  edm::Handle<reco::CandidateView> leptons;               
+  iEvent.getByToken (leptonCollectionTag_, leptons);       
+
+  edm::Handle<reco::JetCorrector> correctorL1L2L3Res;
+  iEvent.getByToken(tagL1L2L3ResCorrector_, correctorL1L2L3Res);
+
+  edm::Handle<reco::JetCorrector> correctorL1;
+  iEvent.getByToken(tagL1Corrector_, correctorL1);
+  
+  edm::Handle<reco::JetTagCollection> bTagHandle;
+  iEvent.getByToken(bTagCollectionTag_, bTagHandle);
+  const reco::JetTagCollection & bTags = *(bTagHandle.product());
+
+  edm::Handle<PFCollection> pfCandidates;
+  iEvent.getByToken(pfCandidates_, pfCandidates);
+
+  edm::Handle<reco::VertexCollection> PVs;
+  iEvent.getByToken(vertexes_, PVs);
+  reco::VertexRef PV(PVs.id());
+  reco::VertexRefProd PVRefProd(PVs);
+
+  //Output
+  std::vector<double > ptratio,ptrel,nchargeddaughers,btagcsv;
```

Comment:
it does not work because the template wants to store a double and then it is not consisted with what is declared from list 110. 
Just made a PR to your branch so as you can include the fix 
https://github.com/folguera/MuonAnalysis-TagAndProbe/pull/1